### ISSUE TITLE
Critical build error

### ISSUE THREAD

**Issue by super-kamil** [strapi/strapi#7142]
**Describe the bug**
The build does not work anymore as expected. .env is ignored and the url is wrong. I can't open admin anymore.

**Steps to reproduce the behavior**

1. update from 3.0.5 to 3.0.6
2. npm run build
3. npm run develop
4. open strapi admin

**Expected behavior**
Admin should fetch http://192.168.87.40:1337/users-permissions/init but it tries to get http://localhost:1337/users-permissions/init

**Code snippets**
server.js
``` 
module.exports = ({ env }) => ({
  host: env('HOST', '0.0.0.0'),
  port: env.int('PORT', 1337),
  url: env('BASEURL', 'http://localhost:1337'),
  env: env('ENV')
});
``` 

.env
``` 
BASEURL="http://192.168.87.40:1337/"
``` 

**System**

- Node.js version: 12.14.1
- NPM version: 6.13.1
- Strapi version: 3.0.6
- Database: mongodb
- Operating system: windows / ubuntu (reproducible on 3 different systems)

Going back to 3.0.5 does not work. All my systems are now broken.

**Comment by super-kamil** [comment]
just found an workaround:

`BASEURL="http://192.168.87.40:1337/" npm run build`

**Comment by aksdevac** [comment]
See issue #7059, its already fixed in `3.1` please update.

**Comment by super-kamil** [comment]
ok thanks. I do not plan to update to 3.1. yet so i have to live with the workaround.
